test/libsolutil/DominatorFinderTest.cpp

```diff
@@ -233,13 +238,21 @@ BOOST_AUTO_TEST_CASE(immediate_dominator_1)
 			{"C", {"G"}},
 			{"D", {"E"}},
 			{"G", {"H"}}
+		},
+		{ // Dominance frontier
+			{"C", {"F", "D"}},
+			{"D", {"F"}},
+			{"E", {"F"}},
+			{"G", {"F"}},
+			{"H", {"F"}}
 		}
 	);
 
 	TestDominatorFinder dominatorFinder(*test.entry);
 	BOOST_TEST(dominatorFinder.immediateDominators() == test.expectedImmediateDominators);
 	BOOST_TEST(dominatorFinder.dfsIndexById() == test.expectedDFSIndices);
```

Comment:
This is not related to this PR, but I have a feeling that testing equality of DFS indices is too strong a requirement.
DFS order is not unique, it depends on the order of processing the outgoing edges. So If I change the implementation of the DFS algorithm (specifically if the way how edges are enumerated changes), everything regarding the dominators should still work correctly, but these tests could break at the same time.

Reply:
Indeed. I added it as a sanity check while implementing the Dominator algorithm, but we could remove it.